crmsh/utils.py

```diff
@@ -2037,33 +2077,17 @@ def detect_cloud():
     * google-cloud-platform
 
     """
-    global _ip_for_cloud
-
     if not is_program("dmidecode"):
         return None
-    rc, system_version = get_stdout("dmidecode -s system-version")
-    if re.search(r".*amazon.*", system_version) is not None:
+    aws = detect_aws()
+    if aws:
         return "amazon-web-services"
-    if rc != 0:
-        return None
-    rc, system_manufacturer = get_stdout("dmidecode -s system-manufacturer")
-    if rc == 0 and "microsoft corporation" in system_manufacturer.lower():
-        # To detect azure we also need to make an API request
-        result = _cloud_metadata_request(
-            "http://169.254.169.254/metadata/instance/network/interface/0/ipv4/ipAddress/0/privateIpAddress?api-version=2017-08-01&format=text",
-            headers={"Metadata": "true"})
-        if result:
-            _ip_for_cloud = result
-            return "microsoft-azure"
-    rc, bios_vendor = get_stdout("dmidecode -s bios-vendor")
-    if rc == 0 and "Google" in bios_vendor:
-        # To detect GCP we also need to make an API request
-        result = _cloud_metadata_request(
-            "http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ip",
-            headers={"Metadata-Flavor": "Google"})
-        if result:
-            _ip_for_cloud = result
-            return "google-cloud-platform"
+    azure = detect_azure()
+    if azure:
+        return "microsoft-azure"
```

Comment:
It would be nice to at some point have this constant as constant variables, and even some kind of enum.
For future improvements I guess

Reply:
Changed as constants
Thanks!